server/public/model/websocket_message.go

```diff
@@ -104,6 +104,17 @@ type WebsocketBroadcast struct {
 	// ReliableClusterSend indicates whether or not the message should
 	// be sent through the cluster using the reliable, TCP backed channel.
 	ReliableClusterSend bool `json:"-"`
+
+	// BroadcastHooks is a slice of hooks IDs used to process events before sending them on individual connections. The
+	// IDs should be understood by the WebSocket code.
+	//
+	// This field should never be sent to the client.
```

Comment:
I see you took the other route. Not against it but we are making it "insecure by default" this way. If it helps keeping complexity low it's acceptable I'd say, but something to keep in mind.

Reply:
Yeah, I ended up going with this because it seems to work, and I've spent far longer on this than I'd expected to originally. As cool as the hook idea that I took from Agniva's other PR is, this kind of blew up in complexity, so I needed to find something that works to ship. In retrospect, I probably should've just gone with sending individual events to mentioned users to fix the original bug without adding all this.

As for making it more secure, I still think the best way to go about that would be to have two separate methods on the event for serializing it for server-server transmission and server-client transmission, but since this has already gotten so big, I didn't want to have to change that whole interface